packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -187,27 +187,26 @@ - (instancetype)initWithURL:(NSURL *)url
   return [self initWithPlayerItem:item frameUpdater:frameUpdater];
 }
 
+// Fix the transform for the given track accordingly.
+// See also: https://stackoverflow.com/a/64161545/13235500
 - (CGAffineTransform)fixTransform:(AVAssetTrack *)videoTrack {
-  CGAffineTransform transform = videoTrack.preferredTransform;
-  // TODO(@recastrodiaz): why do we need to do this? Why is the preferredTransform incorrect?
-  // At least 2 user videos show a black screen when in portrait mode if we directly use the
-  // videoTrack.preferredTransform Setting tx to the height of the video instead of 0, properly
-  // displays the video https://github.com/flutter/flutter/issues/17606#issuecomment-413473181
-  if (transform.tx == 0 && transform.ty == 0) {
-    NSInteger rotationDegrees = (NSInteger)round(radiansToDegrees(atan2(transform.b, transform.a)));
-    NSLog(@"TX and TY are 0. Rotation: %ld. Natural width,height: %f, %f", (long)rotationDegrees,
-          videoTrack.naturalSize.width, videoTrack.naturalSize.height);
-    if (rotationDegrees == 90) {
-      NSLog(@"Setting transform tx");
-      transform.tx = videoTrack.naturalSize.height;
-      transform.ty = 0;
-    } else if (rotationDegrees == 270) {
-      NSLog(@"Setting transform ty");
-      transform.tx = 0;
-      transform.ty = videoTrack.naturalSize.width;
-    }
+  CGAffineTransform t = videoTrack.preferredTransform;
+  CGSize size = videoTrack.naturalSize;
+  t.tx = 0;
+  t.ty = 0;
+  if (t.a == -1) {
+    t.tx = size.width;
+  }
+  if (t.b == -1) {
+    t.ty = size.width;
+  }
+  if (t.c == -1) {
+    t.tx = size.height;
+  }
+  if (t.d == -1) {
+    t.ty = size.height;
```

Comment:
I am a bit iffy about this part. 
1. Will there be scale? if we scale, these numbers won't be just 0/1/-1 anymore right? If we don't, can you add comment? 
2. I feel it is clearer to explicitly write out the 4 numbers in each cases (like [in the stackoverflow answer](https://stackoverflow.com/questions/64161544/avassettracks-preferredtransform-sometimes-seems-to-be-wrong-how-can-this-be-f/64161545#64161545)). This way we can see clearly what rotation we are dealing with in each case. (Also helpful for unit test purpose)

Reply:
> 1. is it just a rotation matrix or does it include scale transformation too? if we scale, these numbers won't be just 0/1/-1 anymore right? If we don't, can you add comment?

Videos will not be scaled here, we only manipulate the x-axis and y-axis offset. I'll add more details in the comment for the method.

> 2. I feel it is clearer to explicitly write out the 4 numbers in each case

Actually, the current conditions were extracted by these four numbers cases, and I personally think they are more clear to review/read. The reason why they are not clear is that the `a`, `b`, `c`, `d` won't be familiar to developers, but it'll be somehow redundant to explain what are these.

Reply:
I feel it is unclear to extract these conditions. These 4 numbers all together represents one single, independent case, for example, rotating 90 degree is `[cos90, sin90, -cos90, sin90]`. I highly recommend separating them so that we see exactly how many cases this function handles, what are these cases, and how many unit tests to write.

Reply:
It's now handled case by case.